lab-taylor/routes/pic-router.js

```diff
@@ -0,0 +1,96 @@
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+const del = require('del');
+const AWS = require('aws-sdk');
+const multer = require('multer');
+const Router = require('express').Router;
+const createError = require('http-errors');
+const debug = require('debug')('instaclone:pic-router');
+
+const Pic = require('../model/pic.js');
+const Gallery = require('../model/gallery.js');
+const bearerAuth = require('../lib/bearer-auth.js');
+const picRouter = module.exports = Router();
+
+AWS.config.setPromisesDependency(require('bluebird'));
+
+const s3 = new AWS.S3();
+const dataDir = `${__dirname}/../data`;
+const upload = multer({ dest: dataDir });
+
+function s3uploadProm(params) {
+  debug('s3uploadProm');
+  
+  return new Promise((resolve) => {
```

Comment:
s3 will return promises if you chain a `.promise()` on the end of the method call